src/app/components/sections/nested-challenges.tsx

```diff
@@ -478,45 +480,64 @@ const StagesContentSection = ({
   attemptsLoading,
   period,
   setPeriod,
+  challengeModules,
 }: {
   challenge: Course;
   attempts: ChallengeAttempt[];
   attemptsLoading: boolean;
   period: ChallengePeriod;
   setPeriod: React.Dispatch<React.SetStateAction<ChallengePeriod>>;
+  challengeModules: Course[];
 }) => {
+  const { allRepositories } = useStore();
+  const repository = allRepositories.find(
+    (repo) => sluggify(repo?.challenge?.title) === sluggify(challenge.title)
+  );
+
+  const isCompleted = repository?.progress?.status === Status.Completed;
+  const moduleCount = challengeModules.length;
+
   return (
     <div className="flex flex-col p-6 pb-6 w-full">
       <NavigationBlock />
-      <div className="flex justify-between pt-4 pb-6">
-        <section className="flex flex-col gap-1">
-          <p className="text-black text-2xl leading-[36px]">
-            {challenge?.title}
-          </p>
-          <p className="text-grey-secondary-text font-light">
-            {challenge?.action}
-          </p>
-        </section>
-      </div>
+      {isCompleted ? (
```

Comment:
This will always pay the user when they come to the challenge page after they've completed the challenge

Reply:
The user should be paid once only